Clase-02/git-basics.md

```diff
@@ -0,0 +1,850 @@
+
+You have unmerged paths.
+  (fix conflicts and run "git commit")
+  (use "git merge --abort" to abort the merge)
+
+Unmerged paths:
+  (use "git add <file>..." to mark resolution)
+	both modified:   README.md
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+        
+Todo aquello que sea conflictivo y no se haya podido resolver, se marca como "sin fusionar" (unmerged). Git añade a los archivos conflictivos unos marcadores especiales de resolución de conflictos que nos guiarán cuando los abramos manualmente y los editemos para corregirlos. 
+
+Si abrimos los archivos en un editor de texto plano convencional, veremos algo como lo siguiente:
+
+```
+<<<<<<< HEAD (Current Changes)
+Mi repositorio en GitHub
+=======
+Mi primer repositorio en GitHub
+>>>>>>> main (Incomming Changes)
+```
+
+Aquí se nos marca que la versión en HEAD (nuetra versión del directorio de trabajo) contiene lo indicado en la parte superior del bloque y la versión entrante contiene el resto, lo indicado en la parte inferior del bloque. 
+
+De todos modos, para una visualización más clara, lo mejor suele ser usar un editor de código del tipo de [Visual Studio Code](https://code.visualstudio.com/)(VSC), donde veremos algo como lo siguiente:
+
+<div style="text-align:center">
+    <img src="git-basics-images/git-visual-conflicts.png" width="50%">
+</div>
+
+
+Si queremos aceptar los cambios entrantes, simplemente debemos hacer click en el **Accept Incoming Change**.
+Si, en cambio, queremos dejar nuestros cambios, debemos elegir **Accept Current Change**.
+O podemos aceptar ambo, para ello usamos **Accept Both Changes**.
+
+Esto debemos hacerlo para cada cambio. Pero, si ya sabemos que queremos aceptar todos los cambios entrantes o todos los cambios actuales (los que están en nuestro directorio de trabajo) y estamos usando VSC, podemos utilizar la paleta de comandos. Para ello, apretamos `Ctrl+Shift+P` y escribimos _Accept All_. Esto hará que se nos muestren las opciones para resolver todos los conflictos de fusión de la misma forma (aceptando todos los cambios entrantes, todos los actuales o ambos cambios).
+
+Si no estamos usando algún editor de código, para resolver nuestros conflictos (los de código, al menos) deberemos abrir un editor de texto plano (Text Editor, Notepad o el que sea de nuestra preferencia), arremangarnos y buscar cada ocurrencia de conflicto (podemos buscar _HEAD_). Cada vez que encontremos una, tenemos que borrar las líneas que indican _<<<<<<< HEAD (Current Changes)_, _>>>>>>> \<branch> (Incomming Changes)_, _=======_ y las de la porción de código que queremos desestimar. Solo debe quedar el cambio que queremos (el current o el incomming). En el ejemplo anterior, deberíamos dejar o bien _Mi repositorio en GitHub_, o bien _Mi primer repositorio en GitHub_
+
+Una vez hecho esto guardamos el archivo y lo agregamos al stagging (con el ya conocido comando `add`). Y si ejecutamos el comando ```git status```, veremos:
```

Comment:
¿Es stagging o staging?

Reply:
Staging. Ya lo modifiqué y lo cambié a itálicas.